portfolio/src/main/webapp/script.js

```diff
@@ -1,28 +1,13 @@
-// Copyright 2020 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+async function showRandomPhrase() {
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+    const responseFromServer = await fetch('/phrase');
+    const phrases = await responseFromServer.json();
+
+    const phrase = phrases[Math.floor(Math.random() * phrases.length)];
```

Comment:
It seems a little strange to me to have the client get the entire list from the server and then pick a random one to show each time. It seems like either the server should pick a random phrase to send or the client should get the list once and use what it's stored to pick a random one from each time.

Maybe I'm missing something?

Reply:
I think the same haha, originally I picked the random phrase within the servlet, but the JSON step specified this:

"Now that you've learned more about JSON, your goal is to modify your servlet to return a list of messages. The messages are up to you! They could be facts about yourself, or your favorite movie quotes.

Then, modify your JavaScript to parse the response it gets from the server as JSON. Then write JavaScript that takes the list it received from the server, chooses a random message from that list, and then adds it to the page. "

And as you mention it would be better to get the list only once :).